### PR TITLE
Purge cache on app update

### PR DESCRIPTION
**Pull request by jnugh**
First of all, please read [`CONTRIBUTING.md`](https://github.com/mattermost/desktop/blob/master/CONTRIBUTING.md)

---

- [x] Complete [Mattermost Contributor Agreement](http://www.mattermost.org/mattermost-contributor-agreement/)
- [x] Execute `npm run prettify` to format codes
- [x] Update `CHANGELOG.md` and/or `doc/*.md` if it's necessary.
- [x] Write about environment which you tested
  - Ubuntu 16.10

### CODE REVIEW

File: src/common/settings.js [hunk 1]
```diff
@@ -2,6 +2,7 @@
 
 const fs = require('fs');
 const settingsVersion = 1;
+const app = require('electron').app;
```
Comment (yuya-oc) [inline src/common/settings.js]: The js is tested in `test/specs/settings_test.js` by mocha. Please consider not using `require('electron')` here. For example, add an arg for `upgrade()`.

File: src/main.js [hunk 2]
```diff
@@ -254,6 +273,7 @@ allowProtocolDialog.init(mainWindow);
 // This method will be called when Electron has finished
 // initialization and is ready to create browser windows.
 app.on('ready', function() {
+
```
Comment (yuya-oc) [inline src/main.js]: Unexpected blank line.

**Comment by jnugh** [comment]
i rebased my PR to address your the changes mentioned by your code review.

**Comment by jasonblais** [comment]
oThanks @jnugh, @yuya-oc! 

Is there a way for us to test this PR? Or is it something where we'd need to monitor for caching issues once the new desktop app is released?

**Comment by yuya-oc** [comment]
By editting `lastMattermostVersion` of `config.json`, the app would assume that the app version has been changed. So you will see the app starts reloading when launching.

Wndows: `%HOME%\AppData\Roaming\Mattermost\config.json`

**Comment by jnugh** [comment]
You could check if the Cache folder in the same directory gets smaller after a version change...
But it might only be smaller for a few seconds, because when loading the webview cache should build up again.

**Comment by jasonblais** [comment]
Hmm, so I tried with the following steps:

1) Download artifact from: https://circleci.com/gh/mattermost/desktop/747#artifacts
2) Install Mattermost (using the 64-bit Windows installer)
3) Go to `%HOME%\AppData\Roaming\Mattermost\config.json` and modify `lastMattermostVersion` from `3.4.1` to `3.5.0`

It seems the app doesn't detect the change, but keeps the version at `3.4.1`. Is there a step I missed?

I also tried closing and re-opening the app.

**Comment by yuya-oc** [comment]
Tested on Windows 10, and the feature works well.

@jasonblais The version is rewritten when launching the app, so keeping `3.4.1` is correct. The clearing caches works very quickly, so probably you can't see reloading. You can see it by watching `%HOME%\AppData\Roaming\Mattermost\Cache`.

**Comment by jnugh** [comment]
The version number in config.json should change back to 3.4.1 when you
start the application next time. The cache will perform purged before the
version string is written to config.json.

Jason Blais <notifications@github.com> schrieb am Sa., 22. Okt. 2016, 21:02:

**Comment by yangchen1** [comment]
+1 @jnugh! Good work.

**Pull request merged**